### PR TITLE
Add xyz.aguno.CubeTimer

### PR DESCRIPTION
**Pull request by herpiko**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link: ** https://github.com/herpiko/cubetimer
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

File: xyz.aguno.CubeTimer.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+  <id>xyz.aguno.CubeTimer.desktop</id>
```
Comment (hfiguiere) [inline xyz.aguno.CubeTimer.appdata.xml:3]:
```suggestion
  <id>xyz.aguno.CubeTimer</id>
```

Which bit of documentation tells to add `.desktop` here. I'd like to get it fixed.

File: xyz.aguno.CubeTimer.appdata.xml
Comment (hfiguiere) [inline xyz.aguno.CubeTimer.appdata.xml:1]: this appstream file is insufficient.

File: xyz.aguno.CubeTimer.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>MIT</project_license>
+  <description>
+    <p>No description</p>
```
Comment (hfiguiere) [inline xyz.aguno.CubeTimer.appdata.xml:7]: that's not a valid description

File: xyz.aguno.CubeTimer.json [hunk 3]
```diff
@@ -0,0 +1,41 @@
+{
+    "app-id" : "xyz.aguno.CubeTimer",
+    "runtime" : "org.gnome.Sdk",
+    "runtime-version" : "42",
```
Comment (hfiguiere) [inline xyz.aguno.CubeTimer.json:4]: this runtime is EOL. Current is 45.

File: xyz.aguno.CubeTimer.json [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    ],
+    "modules" : [
+        {
+            "name" : "cubetimer",
+            "builddir" : true,
+            "buildsystem" : "meson",
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "file:///home/herpiko/src/CubeTimer"
```
Comment (hfiguiere) [inline xyz.aguno.CubeTimer.json:33]: this definitely will not work.

Also for git you need a commit (+ tag).

File: xyz.aguno.CubeTimer.json [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+                }
+            ]
+        }
+    ],
+    "build-options" : {
+        "env" : {        }
+    }
```
Comment (hfiguiere) [inline xyz.aguno.CubeTimer.json:40]:
```suggestion
    ]
```

File: xyz.aguno.CubeTimer.appdata.xml
Comment (hfiguiere) [inline xyz.aguno.CubeTimer.appdata.xml:1]: this and the .desktop file are in the upstream repository so there is no reason to have them here.

**Comment by herpiko** [comment]
Hi @hfiguiere, thank you for your suggestions. I've fixed it. Please kindly review it again.

**Comment by herpiko** [comment]
bot, build xyz.aguno.CubeTimer

File: xyz.aguno.CubeTimer.json [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    ],
+    "modules" : [
+        {
+            "name" : "cubetimer",
+            "builddir" : true,
+            "buildsystem" : "meson",
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "https://github.com/herpiko/cubetimer.git+a04f8eed59bbda28bb82a60b05288e85a50d6a8b"
```
Comment (bbhtt) [inline xyz.aguno.CubeTimer.json:33]: `tag: ...` + `commit: ...` or at least `commit: ...`

**Comment by bbhtt** [comment]
appadata needs to be fixed upstream:

```
xyz.aguno.CubeTimer.appdata.xml.in: FAILED:
• tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]
• style-invalid         : Not enough <screenshot> tags, minimum is 1
• tag-missing           : <release> required
• tag-missing           : <name> is not present
• tag-missing           : <summary> is not present
• tag-missing           : <url> is not present
Validation of files failed
```

File: xyz.aguno.CubeTimer.json (hunk 6 above)
Comment (hfiguiere) [reply xyz.aguno.CubeTimer.json:33]: this is not the syntax at all.

**Comment by bbhtt** [comment]
This icon does not seem correct https://raw.githubusercontent.com/herpiko/cubetimer/a04f8eed59bbda28bb82a60b05288e85a50d6a8b/data/icons/hicolor/scalable/apps/xyz.aguno.CubeTimer.svg

It's the default icon for an executable.

**Comment by herpiko** [comment]
Hi @bbhtt @hfiguiere , I have updated the JSON file, appdata file and also update the icon as well.

Please kindly review my MR again.

**Comment by herpiko** [comment]
bot, build xyz.aguno.CubeTimer

**Comment by herpiko** [comment]
bot, build xyz.aguno.CubeTimer

**Comment by herpiko** [comment]
bot, build xyz.aguno.CubeTimer

**Comment by herpiko** [comment]
bot, build xyz.aguno.CubeTimer

File: xyz.aguno.CubeTimer.json [hunk 7]
```diff
@@ -0,0 +1,39 @@
+{
+    "app-id" : "xyz.aguno.CubeTimer",
+    "runtime" : "org.gnome.Sdk",
```
[the remark this answers is not in the document]
Comment (herpiko) [reply xyz.aguno.CubeTimer.json:3]: @barthalion should it be `org.gnome.Platform` instead?

**Comment by herpiko** [comment]
Hi @barthalion @bbhtt @hfiguiere I've updated the PR. Please kindly review again my submission.

**Comment by herpiko** [comment]
bot, build xyz.aguno.CubeTimer

**Comment by herpiko** [comment]
bot, build xyz.aguno.CubeTimer

**Comment by herpiko** [comment]
Hi @barthalion @bbhtt @hfiguiere when you have time, please kindly review again my submission.

**Comment by herpiko** [comment]
bot, build xyz.aguno.CubeTimer

**Pull request closed without merging**